sql: properly check fieldno in sql.c (#8425)
tarantool/tarantool: changelogs/unreleased/gh-8418-asserton-on-select-from-_space.md

## bugfix/sql

> * Fixed assertion when selecting from '_space' using 'owner' field in
>   WHERE (gh-8418).

Reviewer: ```suggestion
* Fixed an assertion when selecting from `_space` using the `owner` field in
  `WHERE` (gh-8418).
```

Author: Thank you for the review! I changed changelog a bit.
